lgc/builder/DescBuilder.cpp

```diff
@@ -164,10 +161,11 @@ Value *DescBuilder::CreateGetDescStride(ResourceNodeType concreteType, ResourceN
   const ResourceNode *node = nullptr;
   if (!m_pipelineState->isUnlinked() || !m_pipelineState->getUserDataNodes().empty()) {
     std::tie(topNode, node) = m_pipelineState->findResourceNode(abstractType, descSet, binding);
-    if (!node) {
-      // We did not find the resource node. Return an undef value.
+    // NOTE: Resource node may be DescriptorTexelBuffer, but it is defined as OpTypeSampledImage in SPIRV,
+    // The unused DescriptorSampler may need to be found, in this case we will return undefined value.
+    if (!node && m_pipelineState->findResourceNode(ResourceNodeType::Unknown, descSet, binding).second)
       return UndefValue::get(getInt32Ty());
-    }
+    assert(node && "missing resource node");
```

Comment:
Can we just return `nullptr` here instead of `undef` and have the caller check for that? `nullptr` is a clearer, easier-to-check-for, and harder-to-accidentally-ignore indication of "not found". That is:
```
    // NOTE: Resource node may be DescriptorTexelBuffer, but it is defined as OpTypeSampledImage in SPIRV,
    // In this case, a caller may search for the DescriptorSampler and not find it. We return nullptr in this case and
    // and expect the caller to handle it.
    return node;
```
And the same also below.

This should require some changes elsewhere, but it would be overall cleaner.